### PR TITLE
[RW-8059][RW-8058][risk=no] Fix e2e nightly tests

### PR DESCRIPTION
**Pull request by aweng98**
Description:

**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/main/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: e2e/app/sidebar/genomic-extractions-sidebar.ts [hunk 1]
```diff
@@ -50,10 +51,13 @@ export default class GenomicExtractionsSidebar extends BaseSidebar {
...
-        visible: true,
-        timeout
-      });
+      await Promise.all([
+        this.page.waitForXPath(`${table.getXpath()}//*[@data-icon="check-circle" and @role="img"]`, {
+          visible: true,
+          timeout
+        }),
+        waitWhileLoading(this.page, { includeRuntimeSpinner: true, timeout })
+      ]);
```
Comment (aweng98) [inline e2e/app/sidebar/genomic-extractions-sidebar.ts:60]: Avoid false success if there are multiple rows in Extraction table. Add checks for extraction running spinner to be certain extraction has completed.

File: e2e/tests/nightly/admin/user-profile-admin.spec.ts [hunk 2]
```diff
@@ -503,8 +503,8 @@ describe('User Profile Admin page', () => {
     await userProfileAdminPage.getEmailErrorMessageLink().then((link) => link.click());
     const newTarget = await browser.waitForTarget((target) => target.opener() === page.target());
     const newPage = await newTarget.page();
-    // SUBMIT REQUEST button exists
-    await newPage.waitForXPath('//a[text()="SUBMIT REQUEST"]', { visible: true });
+    // SUBMIT A REQUEST button exists
+    await newPage.waitForXPath('//*[./*[text()="Need an Agreement?"]]/a[text()="SUBMIT A REQUEST"]', { visible: true });
```
Comment (aweng98) [inline e2e/tests/nightly/admin/user-profile-admin.spec.ts:507]: UI changed.

File: e2e/tests/nightly/dataproc-to-gce.spec.ts [hunk 3]
```diff
@@ -13,7 +13,7 @@ describe('Updating runtime compute type', () => {
     await signInWithAccessToken(page);
   });
 
-  const workspaceName = 'e2eDataprocToGceTest';
+  const workspaceName = makeRandomName('e2eDataprocToGceTest');
```
Comment (aweng98) [inline e2e/tests/nightly/dataproc-to-gce.spec.ts:16]: Create new workspace or `runtimePanel.applyChanges()` could fail.

File: e2e/tests/nightly/gce-to-dataproc.spec.ts [hunk 4]
```diff
@@ -13,7 +13,7 @@ describe('Updating runtime compute type', () => {
     await signInWithAccessToken(page);
   });
 
-  const workspaceName = 'e2eGceToDataprocTest';
+  const workspaceName = makeRandomName('e2eGceToDataprocTest');
```
Comment (aweng98) [inline e2e/tests/nightly/gce-to-dataproc.spec.ts:16]: Create new workspace or `runtimePanel.applyChanges()` could fail.

File: e2e/tests/nightly/genomic-extraction-to-vcf.spec.ts [hunk 5]
```diff
@@ -155,11 +156,12 @@ describe('Genomics Extraction Test', () => {
...
-    expect(rowCount).toEqual(1);
+    expect(rowCount).toBeGreaterThanOrEqual(1);
 
-    // Verify dataset name (Column #1 is 'Dataset Name').
-    const tableCellValue = await historyTable.getCellValue(1, 1);
-    expect(tableCellValue).toEqual(datasetName);
+    // Verify dataset name found in table (Column #1 is 'Dataset Name').
+    const rowValues = await Promise.all(range(0, rowCount).map((row) => historyTable.getCellValue(row + 1, 1)));
+    const rowIndex = rowValues.findIndex((v) => v === datasetName);
+    expect(rowIndex).not.toBe(-1);
```
Comment (aweng98) [inline e2e/tests/nightly/genomic-extraction-to-vcf.spec.ts:164]: Code hardening even though a new workspace is to be created in each run.

File: e2e/utils/waits-utils.ts [hunk 6]
```diff
@@ -365,9 +365,12 @@ export async function waitWhileSpinnerDisplayed(
...
   const { timeout = 2 * 60 * 1000, includeRuntimeSpinner = false } = opts;
-  const spinnerCss = `[style*="running spin"], .spinner:empty, [style*="running rotation"]${
-    includeRuntimeSpinner ? '' : ':not([aria-hidden="true"]):not([data-test-id*="runtime-status"])'
-  }`;
+  const spinnerCss =
+    '[style*="running spin"], .spinner:empty, ' +
+    `[style*="running rotation"]${
+      includeRuntimeSpinner ? '' : ':not([aria-hidden="true"]):not([data-test-id*="runtime-status"])'
+    }, ` +
+    `[style*="animation-name: spin"]${includeRuntimeSpinner ? '' : ':not([data-test-id*="extraction-status"])'}`;
```
Comment (aweng98) [inline e2e/utils/waits-utils.ts:373]: Modify css selector to take Extraction to VCF status spinner into consideration.

**Review by calbach — APPROVED** [review]

**Pull request merged**